			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-032

Bug-Reported-by:	Albert Akchurin <ackbeat@gmail.com>
Bug-Reference-ID:	Mon, 11 Dec 2023 20:41:58 +0600
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2023-12/msg00047.html

Bug-Description:

Patch (apply with `patch -p0'):

When printing functions containing coprocesses, the displayed coproc command
has the word COPROC inserted unconditionally, resulting in function bodies
that cannot be re-read as input.

*** ../bash-5.2-patched/print_cmd.c	Mon Aug 21 16:09:44 2023
--- print_cmd.c	Mon Dec 11 15:34:30 2023
***************
*** 357,361 ****
  
  	case cm_coproc:
! 	  cprintf ("coproc %s ", command->value.Coproc->name);
  	  skip_this_indent++;
  	  make_command_string_internal (command->value.Coproc->command);
--- 357,363 ----
  
  	case cm_coproc:
! 	  cprintf ("coproc ");
! 	  if (command->value.Coproc->command->type != cm_simple)
! 	    cprintf ("%s ", command->value.Coproc->name);
  	  skip_this_indent++;
  	  make_command_string_internal (command->value.Coproc->command);

*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 31
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 32
  
  #endif /* _PATCHLEVEL_H_ */
